virl2_client/models/lab.py

```diff
@@ -868,6 +949,59 @@ def _create_interface_local(
             iface._type = iface_type
         return iface
 
+    @check_stale
+    @locked
+    def create_annotation(self, annotation_type: str, **kwargs) -> Annotation:
+        """
+        Create a lab annotation
+
+        :param type: type of annotation (rectangle, ellipse, line or text)
+        :returns: the created annotation
+        """
+        url = self._url_for("annotations")
+
+        # create POST json with default annotation property values
+        # override some values by new, expected ones
+        annotation_data = Annotation.get_default_property_values(annotation_type)
```

Comment:
Let's server provide the defaults.  If that's not possible, we should really consider changing the API.  It makes no sense to require defaults (and probably not just in this API endpoint).

Reply:
Every property for each ann. type is required therefore when creating new annotation, you have to provide all of its properties.

I created map of default values, which are the same default values than the one you get when creating annotation through ui.

I made this so you dont have to insert all (required) property values when creating new annotation through PCL, so now you can just do: lab.create_annotation("rectangle")

Reply:
i'd prefer to have a set of functions which defines the properties for annotation, not defined on the backend but locally in the PCL. This is about abstracting complexity and making the use of the API easier by providing "quality of life" functions for the user of the PCL.

1. `Annotation.get_default_properties(annotation_type)` as Patrik proposes which define sane defaults
2. `Annotation.set_default_properties(annotation_type, defaults)` which provides additional values the user wants for this annotation type, they will be merged with the default properties
3. `Annotation.clear_properties(annotation_type)` which will clear the user properties

When the user then creates annotations, the union of the defaults and the user provided properties and the create function are passed to the backend API call to have a complete set of attributes/properties

Example

"defaults for rect": { bg_color="#FFFFFFFF", border_color="#808080FF",  border_style="solid", border_thickness="1", corner_radius=0, layer=0 }

user calls Annotation.create_rect(0,0, 100, 100)  -> properties taken from defaults
user calls Annotation.create_rect(0,0, 100, 100, border_color="#ff000000") -> red rectangle
user calls
Annotations.set_default_properties(Rectangle, border_color="#00ff0000")
Annotations.create_rect(0,0, 100,100)  -> blue rectangle
Annotations.create_rect(100,100, 100,100)  -> another blue rectangle
Annotations.clear_default_properties(Rectangle)
Annotations.create_rect(200,200, 100,100) -> rectangle is black again

This way, the actual creation ALWAYS provides a complete set of defaults and it's easy for the user to control the properties of annotations without having to repeat themselves for every object created.  And the backend does (and should) not have any say in what is a default... the API wants and gets a complete set of properties for every annotation created.

Reply:
the reset to defaults is clearly something that we don't do in PCL (we don't even have such options in CML maybe except of sync from a node def)

we do not reset annotations to default in UI either

do we really think that this would be useful?  why don't we just delete and create a new annotation instead?

Reply:
While I like the idea of having something like "default properties overlay", in reality its not really bringing that much of a QoL change.
While now you have to always provide your custom defaults as a second argument to "create" method (Lab.create_annotation("rectangle", user_defined_defaults)), with your proposal in-place, it would be only one (Lab.create_annotation("rectangle")).

I dont think saving one dict var with user-defined defaults and sending one extra parameter in each create call is that much of a pain, sometimes more abstraction hurts.

Reply:
Another issue is we run async, which means there's a miniscule chance that before you send the request with expected defaultsA, other method might update these class-level defaults to defaultsB and they will be used for the first call, expecting defaultsA, too.

Therefore sending the (user-defined subset of) defaults explicitly, while not being overly painful, gives you more control over what is happening.